store/mockstore/unistore/tikv/mvcc.go

```diff
@@ -817,8 +829,14 @@ func (store *MVCCStore) prewritePessimistic(reqCtx *requestCtx, mutations []*kvr
 		needConstraintCheck := len(req.PessimisticActions) > 0 && req.PessimisticActions[i] == kvrpcpb.PrewriteRequest_DO_CONSTRAINT_CHECK
 		lockExists := lock != nil
 		lockMatch := lockExists && lock.StartTS == startTS
+		lockConstraintPasses := true
```

Comment:
How about moving `lockConstraintPasses := true` out of the loop?

Reply:
Looks not necessary, I think? The check is done key-wise just like `lockExists` and `lockMatch`, and once it fails, the function returns error immediately.

Reply:
Never mind, just getting used to define global state variable out side loops.